### DISCUSSION TITLE
Add a commit (and push) to a tag

### DISCUSSION THREAD

**Discussion by JJ** [community/community#24929] [Actions]
<p>Apparently, if you filter a GitHub action by tag, what you get is something similar to what happens in <a href="https://github.community/t/action-to-add-a-commit-to-a-pr/165866">pull requests</a>: the repo is in a detached head state and then you can’t commit and push there.<br>
I have tried to simply checkout <code>master</code>, but that does not seem to be available. Since the action I use just <code>push</code>es, I can’t re-work the push so that it does so over <code>origin:HEAD</code> (of course I could, if, pun intended, push comes to shove). I was simply wondering if there would be a way of checking out the repo so that HEAD for the main branch would be available.</p>

**Comment by Simran-B** [comment, marked as answer]
<p><code>actions/checkout@v2</code> does a shallow clone by default, so the history will be grafted and no remote branches will be available. You need to set the <code>fetch-depth</code> option to 0 to get the full history and the branches. Then <code>git checkout master</code> should work.</p>
<p>If the tagged commit is on the master branch, you can then <code>git reset --hard &lt;tag&gt;</code>. Or you could do <code>git checkout &lt;tag&gt; -B master</code> to checkout the tag as a new branch “master”, overwriting (!) the existing local branch. But I’m not sure if that is what you want to do.</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="Simran-B" data-post="2" data-topic="167351">
<div class="title">
<div class="quote-controls"></div>
 Simran-B:</div>
<blockquote>
<p>You need to set the <code>fetch-depth</code> option to 1 to get the full history</p>
</blockquote>
</aside>
<p>I think you mean 0? 1 is the default, 0 is used for „get everything“.</p>
